client/client.go

```diff
@@ -169,25 +181,34 @@ func (c *Client) downloadTargets(role string) error {
 	if err != nil {
 		return err
 	}
-	t := c.local.Targets[role].Signed
-	for _, r := range t.Delegations.Roles {
-		err := c.downloadTargets(r.Name)
-		if err != nil {
-			logrus.Error("Failed to download ", role, err)
-			return err
-		}
-	}
+
 	return nil
 }
 
 func (c Client) GetTargetsFile(roleName string, keyIDs []string, snapshotMeta data.Files, consistent bool, threshold int) (*data.Signed, error) {
-	rolePath, err := c.RoleTargetsPath(roleName, snapshotMeta, consistent)
-	if err != nil {
-		return nil, err
+	// require role exists in snapshots
+	roleMeta, ok := snapshotMeta[roleName]
+	if !ok {
+		return nil, fmt.Errorf("Snapshot does not contain target role")
 	}
-	r, err := c.remote.GetMeta(rolePath, snapshotMeta[roleName].Length)
-	if err != nil {
-		return nil, err
+	hashSha256, ok := hashes["sha256"]
```

Comment:
Actually the error message is just wrong. The hash is always required for TUF validation regardless of whether we're doing consistent or not. Consistent only affects generation of filenames. I changed the error message to indicate we only currently support sha256.